app/controllers/api/generic_works_controller.rb

```diff
@@ -0,0 +1,98 @@
+module API
+  # Adds an endpoint that customizes the update action on nested works
+  class GenericWorksController < ApplicationController
```

Comment:
Having the api for GenericWorksController also enabled us to implement the ajax requests, and the ability to update the relationships (adding children or parents) asynchronously within the edit form.